gradoop-examples/gradoop-examples-operators/src/main/java/org/gradoop/examples/keyedgrouping/KeyedGroupingExample.java

```diff
@@ -0,0 +1,100 @@
+import org.gradoop.examples.keyedgrouping.functions.AgeRoundedTo10;
+import org.gradoop.flink.model.api.functions.AggregateFunction;
+import org.gradoop.flink.model.api.functions.KeyFunction;
+import org.gradoop.flink.model.api.functions.KeyFunctionWithDefaultValue;
+import org.gradoop.flink.model.impl.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.operators.aggregation.functions.count.Count;
+import org.gradoop.flink.model.impl.operators.keyedgrouping.KeyedGrouping;
+import org.gradoop.flink.model.impl.operators.keyedgrouping.labelspecific.LabelSpecificKeyFunction;
+import org.gradoop.flink.util.FlinkAsciiGraphLoader;
+import org.gradoop.flink.util.GradoopFlinkConfig;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.singletonList;
+import static org.gradoop.flink.model.impl.operators.keyedgrouping.GroupingKeys.label;
+import static org.gradoop.flink.model.impl.operators.keyedgrouping.GroupingKeys.labelSpecific;
+import static org.gradoop.flink.model.impl.operators.keyedgrouping.GroupingKeys.property;
+
+/**
+ * A self-contained example on how to use {@link KeyedGrouping} and on how to define and use a custom
+ * key-function.
+ */
+public class KeyedGroupingExample {
+
+  /**
+   * Runs the program on the example data graph.
+   * <p>
+   * This provides an example on how to use the {@link KeyedGrouping} operator with label-specific grouping
+   * as well as a custom (user-defined) key-function.
+   * <p>
+   * Using the social network graph {@link SocialNetworkGraph}, the program will:
+   * <ol>
+   *   <li>load the graph from the given gdl string</li>
+   *   <li>group the graph based on<ul>
+   *     <li>vertices and edges by label</li>
+   *     <li>vertices with label {@code Person} additionally by the rounded {@code birthday} property</li>
+   *     <li>vertices with label {@code Forum} additionally by the {@code title} property</li>
+   *   </ul></li>
+   *   <li>print the resulting graph</li>
+   * </ol>
+   *
+   * @param args arguments (unused)
+   * @see <a href="https://github.com/dbs-leipzig/gradoop/wiki/Unary-Logical-Graph-Operators">
+   * Gradoop Wiki</a>
+   * @throws Exception if something goes wrong
+   */
+  public static void main(String[] args) throws Exception {
+    // create flink execution environment
+    ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+    // create loader
+    FlinkAsciiGraphLoader loader = new FlinkAsciiGraphLoader(GradoopFlinkConfig.createConfig(env));
+
+    // load data
+    loader.initDatabaseFromString(SocialNetworkGraph.getGraphGDLString());
+
+    // load the graph
+    LogicalGraph graph = loader.getLogicalGraph();
+
+    // define the key functions to use
+    Map<String, List<KeyFunctionWithDefaultValue<EPGMVertex, ?>>> labelSpecificKeys = new HashMap<>();
+    labelSpecificKeys.put("Person", singletonList(new AgeRoundedTo10<>()));
```

Comment:
I understand that the usage of `singletonList()` in this specific context makes sense, since the lists only contain a single key function. However, i think that examples serve as introduction to a specific feature for people who might never have used it before. By looking at this example, such a person might get the impression, that only lists containing exactly on key function are supported, since `singletonList` explicitly communicates that. I propose adding a list containing multiple key functions or adding a comment to make clear that multiple key functions are indeed supported.

Reply:
I see. I will add a comment, since grouping on additional data does not make sense for this input graph